SPI-GPIO bitbang:

gpio_get_value() returns 0 or nonzero, but getmiso() expects 0 or 1.
Sanitize the value to a 0/1 boolean.

--mb



Index: linux-2.6.28.5/drivers/spi/spi_gpio.c
===================================================================
--- linux-2.6.28.5.orig/drivers/spi/spi_gpio.c	2009-02-15 16:21:11.000000000 +0100
+++ linux-2.6.28.5/drivers/spi/spi_gpio.c	2009-02-15 16:21:47.000000000 +0100
@@ -114,7 +114,7 @@ static inline void setmosi(const struct 
 
 static inline int getmiso(const struct spi_device *spi)
 {
-	return gpio_get_value(SPI_MISO_GPIO);
+	return !!gpio_get_value(SPI_MISO_GPIO);
 }
 
 #undef pdata
